streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -635,14 +634,10 @@ public void setStandbyUpdateListener(final StandbyUpdateListener standbyListener
     final class StreamStateListener implements StreamThread.StateListener {
         private final Map<Long, StreamThread.State> threadState;
         private GlobalStreamThread.State globalThreadState;
-        // this lock should always be held before the state lock
-        private final Object threadStatesLock;
 
-        StreamStateListener(final Map<Long, StreamThread.State> threadState,
-                            final GlobalStreamThread.State globalThreadState) {
-            this.threadState = threadState;
+        StreamStateListener(final GlobalStreamThread.State globalThreadState) {
```

Comment:
Orthogonal question: should we also make this class a singleton? It's use correctly with a single instance right now but might be good to enforce it additionally to guard against future mistakes?

Reply:
I think it can't be a singleton because we support multiple KafkaStreams instances per process. Having it be a private inner class of `KafkaStreams` feels right to me. I've changed its visibility to private accordingly.

Reply:
Good point!